vid_sdl.c

```diff
@@ -100,7 +100,8 @@ static int MapKey( unsigned int sdlkey )
 {
 	switch(sdlkey)
 	{
-	default: return 0;
+	// sdlkey can be Unicode codepoint for non-ascii keys, which are valid
+	default:                      return sdlkey & SDLK_SCANCODE_MASK ? 0 : sdlkey;
```

Comment:
Pretty sure here you could micro optimise to `return sdlkey & ~SDLK_SCANCODE_MASK;` so the compiler can evaluate it to a single `&` instruction.

Reply:
I’m wondering if there can be `sdlkey`s with *not only* `SDLK_SCANCODE_MASK` bit on.

If yes then `sdlkey & ~SDLK_SCANCODE_MASK` probably will give something other than `0`, which doesn’t correspond to [this desired behavior](https://github.com/DarkPlacesEngine/darkplaces/pull/68#issuecomment-1408736934):

> I think you need to filter out sdlkeys (`vid_sdl.c:104`) with the SDLK_SCANCODE_MASK bit on, and still return 0 for those.

Reply:
> I’m wondering if there can be sdlkeys with not only SDLK_SCANCODE_MASK bit on.

You may be correct, but unfortunately it seems there's no documentation of this available currently.  I can find examples of people using `return sdlkey & ~SDLK_SCANCODE_MASK;` but that doesn't mean it's correct so I'll just leave it as is.